src/ark/segmentation/fiber_segmentation.py

```diff
@@ -294,9 +353,147 @@ def segment_fibers(data_xr, fiber_channel, out_dir, fov, blur=2, contrast_scalin
+        tile_length (int):
+            length of tile size, must be a factor of the total image size (default to 512)
+        save_tiles (bool):
+            whether to save cropped images (default to False)
+
+    Returns:
+        tuple (pd.DataFrame, pd.DataFrame):
+         - returns the both fov and tile stats
+    """
+
+    io_utils.validate_paths(fibseg_dir)
+    # this makes sure tile length is a factor of 1024 and 2048
+    if 1024 % tile_length != 0:
+        raise ValueError("Tile length must be a factor of the minimum image size.")
+
+    save_dir = os.path.join(fibseg_dir, f'tile_stats_{tile_length}')
+    fovs = np.unique(fiber_object_table.fov)
+    fov_stats, tile_stats = [], []
+    fov_alignment, fov_avg_length, fov_pixel_density, fov_fiber_density = [], [], [], []
+
+    # get fov level and tile level stats for each image
+    for fov in fovs:
+        fov_fiber_img = io.imread(os.path.join(fibseg_dir, fov + '_fiber_labels.tiff'))
+        fov_length = fov_fiber_img.shape[0]
+        fov_table = fiber_object_table[fiber_object_table.fov == fov]
+
+        # fov level stats
+        fov_align_scores = fov_table['alignment_score'].values
+        fov_alignment.append(np.mean(fov_align_scores[~np.isnan(fov_align_scores)]))
+        fov_avg_length.append(np.mean(fov_table['major_axis_length'].values))
+        fov_p_density, fov_p_density = calculate_density(fov_table, fov_length**2)
+        fov_pixel_density.append(fov_p_density)
+        fov_fiber_density.append(fov_p_density)
+
+        # tile level stats
+        t_alignment, t_length, t_pixel_density, t_fiber_density = [], [], [], []
+        fov_list, tile_x, tile_y = [], [], []
+
+        # create tiles based on provided tile_length
+        for i in range(int(fov_length / tile_length)):
+            y_range = (i*tile_length, (i + 1) * tile_length)
+            for j in range(int(fov_length / tile_length)):
+                x_range = (j * tile_length, (j + 1) * tile_length)
+                fov_list.append(fov)
+                tile_x.append(x_range[0])
+                tile_y.append(y_range[0])
+
+                tile_fiber_img = fov_fiber_img[y_range[0]:y_range[1], x_range[0]:x_range[1]]
+                tile_fiber_img[tile_fiber_img > 0] = 1
+                if save_tiles:
+                    if not os.path.exists(os.path.join(save_dir, fov)):
+                        os.makedirs(os.path.join(save_dir, fov))
+                    io.imsave(os.path.join(save_dir, fov, f'tile_{y_range[0]},{x_range[0]}.tiff'),
+                              tile_fiber_img, check_contrast=False)
+
+                tile_table = fov_table[np.logical_and(
+                    fov_table['centroid-0'] >= y_range[0], fov_table['centroid-0'] < y_range[1])]
+                tile_table = tile_table[np.logical_and(
+                    tile_table['centroid-1'] >= x_range[0], tile_table['centroid-1'] < x_range[1])]
+
+                if len(tile_table) < 5:
```

Comment:
Would be helpful to include a comment on why 5 is the threshold. Also, should we make this a parameter instead of hard-coded?